domains/client/domain-operator/src/domain_bundle_proposer.rs

```diff
@@ -323,6 +328,28 @@ where
             }
         }
 
+        // As a final check, call `check_extrinsics_and_do_pre_dispatch` with all extrinsics,
+        // this is consistent with ER derivation and FP verification
+        if let Err(CheckExtrinsicsValidityError {
```

Comment:
unfortunate that we have to redo these checks. This will be time consuming redoing all the checks again.
We essentially added this because FP does the same check is it ?

Reply:
Right, just in case there are any inconsistencies that may lead to FP and slashing.